### ISSUE TITLE
[MAS4.2.1] Toggle is reported as a checkbox to screenreaders

### ISSUE THREAD

**Issue by connorjsmith** [microsoft/fast#2180]
**What are the steps to reproduce the issue?**
Open a screen reader and navigate to a `<Toggle>`
Observe that it is treated as a checkbox.

**What behavior did you expect?**
Should be read as a "toggle"

**Is there any additional context?**
I believe we should be treating these controls as buttons with the `aria-pressed` attribute
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by chrisdholt** [comment]
Thanks @connorjsmith - I think this needs discussion, it is indeed checkbox functionality and by design. Can you elaborate on why you think `aria-pressed` is a better experience than checked for an control which only provides boolean states (true or false essentially?

**Comment by connorjsmith** [comment]
Looking at the definition of a `switch`, that may actually be more appropriate for this control.

I think the issue is that screen readers say "checkbox checked", instead of a more semantic "toggle on/off".

**Comment by chrisdholt** [comment]
I agree with the Switch definition. The UI only supports true/false values - I think we should update the role of this to switch. That would enable us to keep the rest of the control/api basically the same from what I can see. That would allow us to fix this as a non-breaking change: https://www.w3.org/TR/wai-aria-1.1/#switch

**Comment by chrisdholt** [comment]
Including offline context from @roryfa: 

NVDA and Narrator handle each control as the following:
```
Checkbox:
E.g. <button role="switch" aria-checked="true" />
Narrator: “<name>, checkbox, unchecked”
NVDA: “<name>, checkbox, not checked”
 
Switch:
E.g. <button role="switch" aria-checked="true" />
Narrator: “<name>, checkbox, unchecked”
NVDA: “<name>, checkbox, not checked”
 
Toggle button:
E.g. <button aria-pressed="true"/>
Narrator: “<name>, button, off”
NVDA: “<name>, toggle button, not pressed”
```
Based on this, the recommendation is that we use “Toggle button” since “Switch” is unfortunately interpreted by both screen readers as a checkbox.

**Comment by chrisdholt** [comment]
Still valid and active - slated for next breaking change

**Comment by chrisdholt** [comment]
closed with #3517
